packages/experimental/surface/src/framework/Plugin.ts

```diff
@@ -6,7 +6,10 @@ import { FC, PropsWithChildren } from 'react';
 
 export type PluginProvides<TProvides> = TProvides & {
   context?: FC<PropsWithChildren>;
-  component?: <P extends PropsWithChildren = PropsWithChildren>(datum: any, props?: Partial<P>) => FC<P>;
+  component?: <P extends PropsWithChildren = PropsWithChildren>(
+    datum: any,
+    props?: Partial<P>,
+  ) => FC<PropsWithChildren<{ data: any }>> | undefined | null;
```

Comment:
I might be missing some knowledge, I'm not sure how to declare a type for this generic signature in the call to `definePlugin` – how is that supposed to work?

Either way, `P` is not useful here since `Surface` does not render the `FC` using `P`, so I gave it the props it does render the `FC` with and also permit (some) falsy values since `Surface` also recognizes those.